automation/tests/authentication.spec.ts

```diff
@@ -1,17 +1,150 @@
+  })
 })
 
-test('Guest page loaded', async ({ page }) => {
-  const pageTitle = page.getByText('Elevate Your Mood, One Track at a Time');
-  // const pageTitle2 = page.locator('.description h1');
-  await expect(pageTitle).toHaveText('Elevate Your Mood, One Track at a Time');
+test.describe('@VB-10 - Login. Input validation popup messages', () => {
+   test('@Smoke - Successful login', async ({page}) => {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.fillLoginData(process.env.EMAIL!, process.env.PASSWORD!);
+    await authenticationPage.accountLoginButton.click();
+    await expect(page.getByAltText("An icon of a person's profile view")).toBeVisible();
+  });
+  test('Missing password login - valid email, no password', async ({page})=> {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.fillLoginData(process.env.EMAIL!, loginCredentials.PASSWORD_MISSING);
+    await authenticationPage.accountLoginButton.click();
+    await expect(authenticationPage.popupMessage).toBeVisible();
+    await expect(authenticationPage.popupMessage).toHaveText("You forgot to provide your password!");
+  });
+  test('@Smoke - Invalid login - valid email, invalid password', async ({page})=> {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.fillLoginData(process.env.EMAIL!, loginCredentials.PASSWORD_INVALID);
+    await authenticationPage.accountLoginButton.click();
+    await expect(authenticationPage.popupMessage).toBeVisible();
+    await expect(authenticationPage.popupMessage).toHaveText("Wrong login data! Check your credentials");
+  });
+  test('@Smoke - Invalid login - invalid email, valid password', async ({page})=> {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.fillLoginData(loginCredentials.EMAIL_INVALID, process.env.PASSWORD!);
+    await authenticationPage.accountLoginButton.click();
+    await expect(authenticationPage.popupMessage).toBeVisible();
+    await expect(authenticationPage.popupMessage).toHaveText("Wrong login data! Check your credentials");
+  });
+  test('Missing email login - no email, valid password', async ({page})=> {
+    await authenticationPage.loginButton.click();
+    await authenticationPage.fillLoginData(loginCredentials.EMAIL_MISSING, process.env.PASSWORD!);
+    await authenticationPage.accountLoginButton.click();
+    await expect(authenticationPage.popupMessage).toBeVisible();
+    await expect(authenticationPage.popupMessage).toHaveText("You forgot to provide your email!");
+  });
 });
 
-test('Login Button redirects to login page', async ({page})=>{
-  const loginButton = page.getByRole('link', {name: 'Login'});
-  await loginButton.click();
+test.describe('@VB-11 - Registration form. Input validation popup messages', () => {
+    test('@Smoke - Verify if new account created successfully', async ({page})=> {
+    await authenticationPage.navigateToRegistrationForm();
```

Comment:
Lines 86-87 also should be in a function in a page level as they repeat in 5 tests.